### DISCUSSION TITLE
Public Page - URL Requires #?

### DISCUSSION THREAD

**Discussion by Shadowfita** [Budibase/budibase#6260] [Help]
Hi,

I just wanted to confirm for my own awareness.

I seem to have issues accessing public pages, unless the app name is suffixed with a hash, is that to be expected? I understand having some kind of identifier to differentiate between authenticated and public, but it could be confusing for the end user.

E.G

Authenticated - Will return "staff" homepage, but public redirects to login page: http://hostname:10000/app/applications/
Public - Will return public user homepage: http://hostname:10000/app/applications#/

Thanks!

**Comment by melohagan** [comment]
Hi @Shadowfita 

The hash is simply a result of the hash based routing that is used.
I found a blog that may be of interest: https://blog.bitsrc.io/using-hashed-vs-nonhashed-url-paths-in-single-page-apps-a66234cefc96

In short though, this is simply the way we handle routing, and has nothing to do with access levels or authentication.

**Reply by Shadowfita** [reply]
Thanks for the response! That's interesting then. Why do you think the noted behaviour may be occurring?

**Reply by Shadowfita** [reply]
Hi @melohagan 

Just following up on this. I have a public "home page" ( / ), and a public page following this url schema: /applications/new/:id . 

If I navigate to http://instance/myapplication/applications/new/:id it will show me the public "home page", but if I navigate to http://instance/myapplication#/applications/new/:id (note the hash), it will load the new application page as expected.

Thanks for your help!

**Reply by melohagan** [reply]
It may be because you're not supplying a value for the URL variable? Is it still an issue if you replace the `:id` with a valid id?

**Reply by Shadowfita** [reply]
> It may be because you're not supplying a value for the URL variable? Is it still an issue if you replace the `:id` with a valid id?

Yep, issue is still present. I was just using ':id' for examples sake.

**Reply by melohagan** [reply]
Great - thanks for letting us know!
